implicit-hie.cabal

```diff
@@ -16,41 +9,68 @@ bug-reports:    https://github.com/Avi-D-coder/implicit-hie/issues
 author:         Avi Dessauer
 maintainer:     avi.the.coder@gmail.com
 copyright:      2020
-license:        BSD3
+license:        BSD-3-Clause
 license-file:   LICENSE
 build-type:     Simple
 extra-source-files:
     README.md
     ChangeLog.md
+    test/testdata/cabal-helper/**/*.hs
+    test/testdata/cabal-helper/**/*.cabal
+    test/testdata/cabal-helper/**/*.project
 
 source-repository head
   type: git
   location: https://github.com/Avi-D-coder/implicit-hie
 
+flag cabalHelper
+  Description: Enable Cabal-Helper dependencies
+  Default:     True
+  Manual:      True
+
+common cabal-helper
+  if flag(cabalHelper)
+    cpp-options: -DCABAL_HELPER_SUPPORT
+
 library
+  import: cabal-helper
   exposed-modules:
       Hie.Cabal.Parser
       Hie.Locate
+      Hie.Logger
       Hie.Yaml
+
+  if flag(cabalHelper)
+    exposed-modules:
+      Hie.CabalHelper
+    build-depends:
+      cabal-helper >= 1.1 && <1.2
+
   other-modules:
       Paths_implicit_hie
   autogen-modules:
       Paths_implicit_hie
   hs-source-dirs:
       src
-  ghc-options: -Wall -Wincomplete-record-updates -Wincomplete-uni-patterns -fno-warn-unused-imports -fno-warn-unused-binds -fno-warn-name-shadowing -fwarn-redundant-constraints
+  ghc-options: -Wall -Wincomplete-record-updates -Wincomplete-uni-patterns -fwarn-unused-imports -fwarn-unused-binds -fwarn-name-shadowing -fwarn-redundant-constraints
   build-depends:
-      attoparsec >= 0.13
+      aeson
+    , attoparsec >= 0.13
     , base >=4.7 && <5
+    , containers
     , directory >= 1.3
     , filepath >= 1.4
     , filepattern >= 0.1
+    , hie-bios >=0.5 && <0.6
+    , hslogger
     , text >= 1.2
     , transformers >= 0.5
+    , process
```

Comment:
All these new deps should be behind the flag.

Reply:
Done.
I argue though that the hslogger dependency is fine, since implicit-hie needs a logger anyway.

Reply:
Sounds good